basyx.aasxfileserver/basyx.aasxfileserver-http/src/main/java/org/eclipse/digitaltwin/basyx/aasxfileserver/http/AASXFileServerHttpApiController.java

```diff
@@ -0,0 +1,173 @@
+import org.eclipse.digitaltwin.aas4j.v3.model.Submodel;
+import org.eclipse.digitaltwin.basyx.aasxfileserver.AASXFileServer;
+import org.eclipse.digitaltwin.basyx.aasxfileserver.model.PackageDescriptionList;
+import org.eclipse.digitaltwin.basyx.aasxfileserver.pagination.GetPackageDescriptionResult;
+import org.eclipse.digitaltwin.basyx.core.exceptions.ElementDoesNotExistException;
+import org.eclipse.digitaltwin.basyx.core.pagination.CursorResult;
+import org.eclipse.digitaltwin.basyx.core.pagination.PaginationInfo;
+import org.eclipse.digitaltwin.basyx.http.Base64UrlEncodedIdentifier;
+import org.eclipse.digitaltwin.basyx.http.pagination.Base64UrlEncodedCursor;
+import org.eclipse.digitaltwin.basyx.http.pagination.PagedResult;
+import org.eclipse.digitaltwin.basyx.http.pagination.PagedResultPagingMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.awt.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * API Controller for the AASX File Server HTTP API
+ *
+ * @author zielstor, fried
+ */
+@RestController
+public class AASXFileServerHttpApiController implements AASXFileServerHttpApi {
+
+    private final AASXFileServer aasxFileServer;
+
+    @Autowired
+    public AASXFileServerHttpApiController(AASXFileServer aasxFileServer) {
+        this.aasxFileServer = aasxFileServer;
+    }
+
+    @Override
+    public ResponseEntity<Void> deleteAASXByPackageId(Base64UrlEncodedIdentifier packageId) {
+        boolean exists = doesPackageExist(packageId);
+        if(!exists){
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+        aasxFileServer.deleteAASXByPackageId(packageId.getIdentifier());
+        return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    }
+
+    @Override
+    public ResponseEntity<Resource> getAASXByPackageId(Base64UrlEncodedIdentifier packageId) {
+        try {
+            InputStream aasxPackage = aasxFileServer.getAASXByPackageId(packageId.getIdentifier());
+            InputStreamResource resource = new InputStreamResource(aasxPackage);
+
+            return ResponseEntity.ok()
+                    .contentType(MediaType.parseMediaType("application/asset-administration-shell-package"))
+                    .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=test.aasx")
+                    .body(resource);
+        } catch(ElementDoesNotExistException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
```

Comment:
Is this expection not being handled by the ControllerAdvice?

Reply:
Yes you're right